modules/datadict/jsx/dataDictIndex.js

```diff
@@ -38,23 +40,94 @@ class DataDictIndex extends Component {
    * Called by React when the component has been rendered on the page.
    */
   componentDidMount() {
-    this.fetchData()
-      .then( () => this.setState({isLoaded: true}));
+      // Load the field options. This comes from a separate request than the
+      // table data stream. Once the fieldOptions are loaded, we set isLoaded
+      // to true so that the page is displayed with the data that's been
+      // retrieved.
+      fetch(this.props.fieldsURL, {credentials: 'same-origin'})
+          .then((resp) => resp.json())
+          .then((data) => this.setState({fieldOptions: data, isLoaded: true}))
+        .catch((error) => {
+            this.setState({error: true});
+            console.error(error);
+        });
+    this.fetchData();
   }
 
   /**
    * Retrive data from the provided URL and save it in state
    *
    * @return {object}
    */
-  fetchData() {
-    return fetch(this.props.dataURL, {credentials: 'same-origin'})
-        .then((resp) => resp.json())
-        .then((data) => this.setState({data}))
-        .catch((error) => {
-            this.setState({error: true});
-            console.error(error);
-        });
+    fetchData() {
```

Comment:
Is there a way to place this function somewhere where other modules could reuse it?

Reply:
I don't know.. is there? We didn't for the previous fetchData that's often the same between modules, so I don't know of anywhere to do it for this one.

Reply:
Not at the moment - but I guess we could start a generic Utility class for certain re-useable javascript functions? Not sure what best practices are in that regard though. @maltheism do you know of a good solution to not have to duplicate this in every module?

Reply:
@HenriRabalais I like the idea you have for a generic Utility include that would have functions such as the one @driusan wrote. Here is a blog post that shows three popular approaches: https://christhedeveloper.com/react-resuable-functions/
I like the second approach which is what I believe you're describing in your comment.

Reply:
@maltheism thanks for the link! I personally have been using the 2nd approach as I'm finding it the simplest. I guess the difficulty would be more so in ensuring that the library file remains organized.

Reply:
@HenriRabalais it would be awesome for organization. I like the idea!

Reply:
So what's the consensus on what to do here? Where should I move the function?

Reply:
@driusan maybe it should be discussed at Loris meeting. The idea of having a new Utility include for this function and other useful functions we can import for frontend development.

Reply:
@maltheism sounds good to me!

Reply:
It's been moved to jslib, can you re-review?